### PR TITLE
0.5

### PR DESCRIPTION
**Pull request by boblail**
# Neat.js v0.5

This refactor:
- Extracts Renderers from `Neat.CollectionEditor` and provides three:
  - `Neat.Renderer.Basic` (which renders everything without pagination)
  - `Neat.Renderer.Paginated` (which renders collections in pages)
  - `Neat.Renderer.InfiniteReveal` (which renders a certain number of items at a time and renders more as the user scrolls down until the whole in-memory collection is rendered)
- Optimizes re-renders at the collection-level by re-rendering only those rows that have changed
- Removes several features from Neat.js:
  - Built-in support for sorting (We recommend using `Backbone.Collection.comparator` instead)
  - Built-in support for adding `.alt` css to rows (We recommend using `:nth-child` instead)
- Switched pagination to **off**, by default, on `Neat.CollectionEditor` (i.e. switched the default renderer from `Neat.Renderer.Paginated` to `Neat.Renderer.Basic`)

### CODE REVIEW

File: vendor/assets/javascripts/neat/renderer/basic.coffee [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  _visibleModels: ->
+    @collection.toArray()
+
+  _indexOfViewForModel: (model)->
+    _.findIndex @views, (view)-> view.model.cid is model.cid
+
+
+
+  _collectionHasBeenReset: ->
+    @render()
```
Comment (kobsy) [inline vendor/assets/javascripts/neat/renderer/basic.coffee]: Should be `@_render()` (Also in the next 4 methods)

File: vendor/assets/javascripts/neat/basic_collection_renderer.coffee [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    # balances could be updated very rapidly. Redrawing the
+    # page will slow things down dramatically.
+    #
+    # Instead, redraw the page 500ms after a model has changed.
+    #
+    # This allows us to wait for activity to die down
+    # and to redraw the page when it's more likely the system
+    # has settled into new state.
+    #
+    @delayedRerender = new Lail.DelayedAction(_.bind(@_modelHasBeenChanged, @), delay: 500)
```
Comment (ben5516) [inline vendor/assets/javascripts/neat/basic_collection_renderer.coffee]: If multiple models change, does this `debouncer` call render multiple times or just once after the final change that occurs within the `delay` time?

Comment (boblail) [inline vendor/assets/javascripts/neat/basic_collection_renderer.coffee]: It gets removed in a later commit.

In this one, it's just being moved from `Neat.PaginatedCollectionRenderer` (where it was extracted from `Neat.CollectionEditor`)

Comment (boblail) [inline vendor/assets/javascripts/neat/basic_collection_renderer.coffee]: Although, to answer your question: just once after the final change (and it rerenders the whole collection)

**Comment by ben5516** [comment]
Chase and I give this a :+1:

**Pull request merged**